### PR TITLE
Sapient bots can play voice lines

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Part of the "throwing stuff at the wall and seeing what happens" series of changes.
We were discussing player bots and there was some small support for "what if bots couldn't talk except to other bots and primarily communicated via emoting and playing their prerecorded lines". 
I coded this but I'll be honest, my heart wasn't in it, especially because we have _already_ released sapient bots who can talk so I'd have to justify taking it away in a better way than "I want to see what would happen", so I'm not doing that (feel free to revert the last commit in this PR if you want to carry that cross yourself).

But I'm keeping the button that plays the prerecorded voice messages.
No these lists _aren't_ alphabetised, that's because I wanted to vaguely order them in order of "this is something the bot regularly says" to "this is something the bot rarely says (because it is rude or mean)". This priority is possibly the inverse of how players will want to use the button, but perhaps they should try being politer.

Additionally this PR ensures that if you put a pAI in a bot then it will be able to speak any languages the pAI could.

## Why It's Good For The Game

Saying "I should have been a plastic surgeon" is iconic medbot behaviour and players should be able to do it.
We don't currently have bot TTS voices which directly sound like the bots themselves to apply, and it might be mildly concerning if we did because I believe at least one of the voices is the IRL voice of a community member with some filters added?

## Changelog

:cl:
add: Player-controlled bots can now play the prerecorded lines associated with that bot.
add: Placing a pAI into a bot temporarily teaches that bot all the langauges the pAI could speak.
/:cl:

### CODE REVIEW

**Comment by EricZilla** [comment]
You could make the mute bots a config option in case people don't want them talking, but I am very against bots being unable to talk, I recently had a very lovely round where I was a cleaningbot on a whiteship who was found and went on space adventures, which really would have been harder to get off the ground since I'm pretty sure cleaning bots don't even have voicelines

**Comment by SomeoneYouProbablyKnow** [comment]
Yeah talkingnis probably the least problematic part of bots

File: code/modules/mob/living/simple_animal/bot/bot_announcement.dm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		if (feedback)
+			owner.balloon_alert(owner, "no announcement system!")
+		return FALSE
+	if (!length(automated_announcements))
+		if (feedback)
+			owner.balloon_alert(owner, "no valid announcements!")
+		return FALSE
+	return TRUE
+
+/datum/action/cooldown/bot_announcement/Trigger(trigger_flags, atom/target)
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/bot/bot_announcement.dm:32]: Cooldown actions shouldn't override `Trigger` for this purpose, they have `Activate`

(In addition, not checking the return value of parent means all the code in `IsAvailable` does nothing)

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by MrMelbert** [comment]
(Also I will now plant a brain worm in you: AI bots controlled go through this action regardless to make sounds regardless, would make it 100% consistent between AI and Human + easier when converting to basic mobs since the behavior is entirely separate)

**Comment by Jacquerel** [comment]
> (Also I will now plant a brain worm in you: AI bots controlled go through this action to make sounds as well - would make it 100% consistent between AI and Human AND easier when converting to basic mobs since the behavior is entirely separate)

I will try, although I will have to figure out how the AI is going to make the UI input.

**Comment by MrMelbert** [comment]
> I will try, although I will have to figure out how the AI is going to make the UI input.

You don't need to go through the UI, ideally you'd have a separate proc like `run_voice_line` which handles the fun part, triggering it as a human goes through the input while AI code can just do `speaker_action.run_voice_line(whatever)` 

Maybe separate out the voice line list into categories for the more complex box, idk. You don't have to do it, it is just a brain worm

**Comment by Dirt-Dweller** [comment]
>add: Placing a pAI into a bot temporarily teaches that bot all the languages the pAI could speak.

Unless I'm missing something, pAIs cant be put into bots anymore.

**Pull request merged**